From 4c7e523baff45d78438194dccaf04fb908678b53 Mon Sep 17 00:00:00 2001
From: Markus Boehme <markus.boehme@chainguard.dev>
Date: Mon, 5 Jan 2026 22:54:26 +0100
Subject: [PATCH] systemd: allow resolved to call networkd resolve hook

With systemd v259, networkd provides a resolve hook that resolved can
call to resolve hostnames that came in via a networkd DHCP lease.

Signed-off-by: Markus Boehme <markus.boehme@chainguard.dev>
---
 policy/modules/system/systemd.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index fc841de9c..3fe5ba1c2 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -1610,6 +1610,7 @@ init_pid_filetrans(systemd_resolved_t, systemd_resolved_var_run_t, dir)
 list_dirs_pattern(systemd_resolved_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 read_files_pattern(systemd_resolved_t, systemd_networkd_var_run_t, systemd_networkd_var_run_t)
 allow systemd_resolved_t systemd_networkd_var_run_t:dir watch_dir_perms;
+allow systemd_resolved_t systemd_networkd_t:unix_stream_socket { connectto };
 
 kernel_dgram_send(systemd_resolved_t)
 kernel_read_net_sysctls(systemd_resolved_t)
-- 
2.52.0

